activerecord/lib/active_record/migration.rb

```diff
@@ -324,28 +324,6 @@ def initialize(name)
   # For a list of commands that are reversible, please see
   # <tt>ActiveRecord::Migration::CommandRecorder</tt>.
   class Migration
-    def self.add_foreign_key(parent, child, cascade = false, foreign_key = parent.to_s.singularize + '_id')
-      constraint_name = "#{child.to_s}_#{foreign_key}"
-      # If an index does not exist one is created with the same name as the constraint.
-      execute "ALTER TABLE #{child} ADD CONSTRAINT #{constraint_name} FOREIGN KEY #{constraint_name}(#{foreign_key}) REFERENCES #{parent}(id)" + (cascade ? " ON DELETE CASCADE" : "")
-    end
-
-    def self.remove_foreign_key(parent,child,remove_index=false )
```

Comment:
looks like add_foreign_key and remove_foreign_key didn't make the transition to the web repo - were these on the "solved in Rails 4" list or similar?